src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/GeneratedCodeContainer.cs

```diff
@@ -103,15 +191,25 @@ internal bool TryGetLinePositionSpan(TextSpan span, out LinePositionSpan linePos
             return false;
         }
 
+        private void TextContainer_TextChanged(object sender, TextChangeEventArgs args)
+        {
+            lock (_setOutputLock)
```

Comment:
The lock here is misleading. This shouldn't take the lock, this needs to be called **with** the lock.